src/packet/route/route.rs

```diff
@@ -283,6 +289,58 @@ impl<'a> ToPayload for RtAttrPacket<'a> {
     }
 }
 
+impl<'a> ToPayload for &'a [&'a Nested<'a>] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        self.iter().fold(0, |pos, pkg| {
+            pkg.payload_add(&mut payload[pos..]);
+            pos + pkg.payload_size()
+        });
+    }
+
+    fn payload_size(&self) -> usize {
+        self.iter().map(|p| p.payload_size()).sum()
+    }
+}
+
+impl<'a> ToPayload for &'a [u8] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        payload[..self.len()].copy_from_slice(self)
+    }
+
+    fn payload_size(&self) -> usize {
+        self.len() + 1
+    }
+}
+
+impl<'a> ToPayload for Nested<'a> {
```

Comment:
I would accept `ToPayload` for `IfInfoPacket`. If you need this enum in your project, I would say you implement it in your project. I have done the same for a project of me.